src/game/objects/Pig.js

```diff
@@ -38,12 +38,12 @@ class Pig extends GameObject {
   }
 
   /**
-   * @param {import('../shared').Collision} _collision
+   * @param {import('../shared').Collision} collision
    * @param {GameObject} target
    */
-  onCollision(_collision, target) {
+  onCollision(collision, target) {
     if (target instanceof SolidTile) {
-      const normal = resolveCollisionsWithSolid(this, target, 0.5);
+      const normal = SolidTile.resolveCollisionWith(this, collision, 0.25);
 
       if (normal.y < 0) {
         this.#isStanding = true;
```

Comment:
Moim zdaniem ten cały kod można przenieść do `SolidTile.onCollision`. W `onCollision` jako drugi argument dostanie obiekt i może spokojnie tam zmieniać jego `transform`, `rigidbody` itp. - ten sam kod dla króla, świnki, boxów i tak naprawdę wszystkiego co nie przechodzi przez ściany. Jeżeli chodzi o tego if'a to można go przenieść do onUpdate i np. sprawdzić czy rigidbody w `y` jest zero?

Reply:
Od takiej implementacji zacząłem, ale nie znalazłem sensownej drogi na przekazanie danych do `#isStanding` stamtąd.

Nie możemy wiązać stania z `rigidbody.velocity.y === 0`, bo, np. w trakcie skoku w najwyższym punkcie v = 0, a nie stoimy + nawet przy swobodnym spadku nie mamy gwarancji, że v = 0 po "wleceniu" w blok pod nami jeśli `restitution !== 0`.

Próbowałem też przez chwilę z `onCollision` w `SolidTile` przekazać ten wektor metodą do świnki, ale to jedynie przeniosło duplikację w inne miejsce. Jakieś rozwiązanie to przenieść `isStanding` do `GameObject`, ale na pewno nie idealne, bo nie w każdym obiekcie będzie potrzebna taka logika - ewentualnie można zrobić klasę dziedziczącą po `GameObject`, która będzie zawierała logikę kolizji z `SolidTile` i `isStanding` (po której dziedziczyłyby np. świnka, król, skrzynki itd., ale taki `SolidTile` czy `Door` już nie).

Chyba że jest jakiś inny, "pewny" predykat na `isStanding` niż ten obecny if.

Reply:
A coś takiego?
```js
onCollision(collision, target) {
  if (target instanceof SolidTile && collision.resolutionVector.y < 0) {
    this.#isStanding = true;
  }
}
```

Reply:
O, to brzmi spoko, rano tak zrobię.

Reply:
Mniej więcej wyszło, jedyne co to ESLint krzyczy o circular dependency i nie wiem jak to sprytnie rozwiązać (bo w tym rozwiązaniu w `Pig.onCollision` musimy mieć dostęp do `SolidTile`, a w `SolidTile.onCollision` do `Pig` pośrednio lub bezpośrednio), także na tę chwilę wyciszyłem, bo pomimo krzyku lintera w praktyce wszystko działa.